### PR TITLE
Bilinear interp op

### PR DESCRIPTION
**Pull request by wangyang59**
Implemented the bilinear interpolation op in Fluid based on @luotao1 #3925 and old paddle's implementation. Added python unittest. Adapted to new Op interface. Responded to original comments.

GPU kernel implemented

### CODE REVIEW

File: paddle/fluid/operators/bilinear_interp_op.cc [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+  BilinearInterpOpMaker(OpProto* proto, OpAttrChecker* op_checker)
+      : OpProtoAndCheckerMaker(proto, op_checker) {
+    AddInput("X",
+             "The input tensor of bilinear interpolation, 4-D with NCHW shape");
+    AddOutput("Out", "The output tensor with the same shape as X");
+    AddAttr<int>("out_h", "output height of bilinear interpolation op.");
+    AddAttr<int>("out_w", "output weight of bilinear interpolation op.");
+    AddComment(R"DOC(
+          Bilinear interpolation is an extension of linear interpolation for 
+          interpolating functions of two variables (e.g. H-direction and W-direction
```
Comment (pkuyym) [inline paddle/fluid/operators/bilinear_interp_op.cc]: exceed 80 columns.

Comment (wangyang59) [reply paddle/fluid/operators/bilinear_interp_op.cc]: Hi @pkuyym , I have reduced the line width down to 80. I have also implemented the GPU kernel. Please let me know if you have any comments.

**Comment by wangyang59** [comment]
@pkuyym @luotao1 Could you please help review the code? Thanks!

File: paddle/fluid/operators/bilinear_interp_op.cu.h [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+    T w2lambda = 1.f - w1lambda;
+
+    T* inPos = &in[outIdH * inputW + channelId * inImgSize + inImgIdy * inImgW +
+                   inImgIdx];
+    const T* outPos = &out[outIdH * outputW + outIdW];
+    atomicAdd(&inPos[0], h2lambda * w2lambda * outPos[0]);
+    atomicAdd(&inPos[wId], h2lambda * w1lambda * outPos[0]);
+    atomicAdd(&inPos[hId * inImgW], h1lambda * w2lambda * outPos[0]);
+    atomicAdd(&inPos[hId * inImgW + wId], h1lambda * w1lambda * outPos[0]);
+  }
```
Comment (qingqing01) [inline paddle/fluid/operators/bilinear_interp_op.cu.h]: The kernel in `.cu.h` can be moved to `.cu` file.  In addition, please note the code style.  

inImgW ->in_img_w in Fluid, this is different from old Paddle.

File: paddle/fluid/operators/bilinear_interp_op.cc [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+  }
+};
+
+class BilinearInterpOpMaker : public framework::OpProtoAndCheckerMaker {
+ public:
+  BilinearInterpOpMaker(OpProto* proto, OpAttrChecker* op_checker)
+      : OpProtoAndCheckerMaker(proto, op_checker) {
+    AddInput("X",
+             "The input tensor of bilinear interpolation, 4-D with NCHW shape");
+    AddOutput("Out", "The output tensor with the same shape as X");
```
Comment (qingqing01) [inline paddle/fluid/operators/bilinear_interp_op.cc]: AddOutput("Out", " (Tensor)  The dimension of output is [N,C, out_h, out_w]");

File: paddle/fluid/operators/bilinear_interp_op.cc [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+    AddAttr<int>("out_h", "output height of bilinear interpolation op.");
+    AddAttr<int>("out_w", "output weight of bilinear interpolation op.");
```
Comment (qingqing01) [inline paddle/fluid/operators/bilinear_interp_op.cc]: Please refer Refer: https://github.com/PaddlePaddle/Paddle/blob/develop/paddle/fluid/operators/lstm_op.cc#L108 for the doc format.

```
AddAttr<int>("out_h", "(int) xxx");
```

File: python/paddle/fluid/tests/unittests/test_bilinear_interp_op.py [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+        self.input_shape = [4, 1, 7, 8]
+        self.out_h = 1
+        self.out_w = 1
+
+
+class TestCase2(TestBilinearInterpOp):
+    def init_test_case(self):
+        self.input_shape = [3, 3, 9, 6]
+        self.out_h = 12
+        self.out_w = 12
```
Comment (qingqing01) [inline python/paddle/fluid/tests/unittests/test_bilinear_interp_op.py]: For more robust testing, better use larger shape.

**Comment by wangyang59** [comment]
@qingqing01, thanks a lot for your review!
I have modified the code according to your suggestions. Currently, I made the largest test size to be around 100, since larger tests would take much longer.  Right now, it takes around 6 seconds to finish the bilinear_interp_op test. Please let me know if you are OK with it.

**Pull request merged**